javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -332,7 +319,7 @@ private Vector(Class<?> type, Object leading, VectorTree<T> middle, Object trail
      */
     public static Vector<Integer> ofAll(int[] array) {
```

Comment:
everything is stored as it is provided: if it was boxed before, it will be stored as such.
If we get a primitive array, we will store it like that.